TheraLang.BLL/Services/UserService.cs

```diff
@@ -0,0 +1,128 @@
+﻿using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using TheraLang.BLL.DataTransferObjects;
+using TheraLang.BLL.Interfaces;
+using TheraLang.DAL.Entities;
+using TheraLang.DAL.UnitOfWork;
+
+namespace TheraLang.BLL.Services
+{
+    public class UserService : IUserService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IFileService _fileService;
+
+        public UserService(IUnitOfWork unitOfWork, IFileService fileService)
+        {
+            _unitOfWork = unitOfWork;
+            _fileService = fileService;
+        }
+
+        public async Task<UserAllDto> GetMyProfile(Guid id)
+        {
+            try
+            {
+                var user = await _unitOfWork.Repository<User>().Get(u => u.Id == id);
+                var userDetails = await _unitOfWork.Repository<UserDetails>().Get(u => u.UserDetailsId == id);
+
+                var detailsMapper = new MapperConfiguration(cfg => cfg.CreateMap<UserDetails, UserAllDto>()).CreateMapper();
+                var userAll = detailsMapper.Map<UserDetails, UserAllDto>(userDetails);
+
+                var mapper = new MapperConfiguration(cfg => cfg.CreateMap<User, UserAllDto>()).CreateMapper();
+                var userInf = mapper.Map<User, UserAllDto>(user);
+                userAll.Email = userInf.Email;
+
+                return userAll;
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"Error when getting user by {nameof(id)}={id}: ", ex);
+            }
+        }
```

Comment:
User email and role already on frontend in token, only need load details.

Reply:
I didnt use it on frontend. It does not metter.

Reply:
You get User entity and map it to UserAllDto only to get user email, which is already contained in jwt we got in request and also stored in local storage in browser. So returning email to a logged in user is redundant

Reply:
Not only for user email. We get also another properties like first_name, last_name, short_inf, img_url ...

Reply:
These fields we get from UserDetails entity. I was talking about line 28, 34-36

Reply:
It's not important. We will make mess in frontend logic without any sense. And also, we already have frontend without this stuff.

Reply:
Ok, how about that?
```
var userDetails = await _unitOfWork.Repository<UserDetails>()
                    .GetAll()
                    .Include(ud => ud.User)
                    .FirstOrDefaultAsync(ud => ud.UserDetailsId == id);
                var detailsMapper = new MapperConfiguration(cfg =>
                    cfg.CreateMap<UserDetails, UserAllDto>()
                        .ForMember(userAllDto => userAllDto.Email,
                            opts => opts.MapFrom(details => details.User.Email))).CreateMapper();
                var userAll = detailsMapper.Map<UserDetails, UserAllDto>(userDetails);
                return userAll;
```

Reply:
ok

Reply:
done